File: python/encode.py
import sys
import M2Crypto
import M2Crypto.Rand
import M2Crypto.RSA
import base64
import json
import os

if len(sys.argv) != 3:
	print sys.argv[0], " payload.json sealed.json"
	sys.exit(1)

public_key = os.path.dirname(os.path.realpath(__file__)) + "/../_shared/public_key.pem"
messagefile = open(sys.argv[1])
data = messagefile.read()
messagefile.close()

public_key = M2Crypto.RSA.load_pub_key(public_key)
randdata = M2Crypto.Rand.rand_bytes(16)
msg = public_key.public_encrypt(randdata,M2Crypto.RSA.pkcs1_padding)
token = base64.b64encode(msg)

cipher='rc4'
iv = ""
enc = M2Crypto.EVP.Cipher(cipher, randdata, iv, 1)
payload = enc.update(data)
payload += enc.final()
payload = base64.b64encode(payload)

j = {'token':token,'payload':payload}

f = open(sys.argv[2],'w')
f.write(json.dumps(j))
f.close()

File: ruby/encode.rb
require 'rubygems'
require 'json'
require 'openssl'
require 'SecureRandom'
require 'base64'

unless ARGV.length == 2
	puts "#$0 payload.json sealed.json"
	exit
end

data = File.read(ARGV[0])

path = File.expand_path(File.dirname(__FILE__)) + '/../_shared/public_key.pem'
public_key = OpenSSL::PKey::RSA.new(File.read(path))
rand = SecureRandom.random_bytes(16)
token = public_key.public_encrypt(rand)

cipher = OpenSSL::Cipher.new('rc4')
cipher.encrypt
cipher.key = rand

payload = cipher.update(data) + cipher.final

sealed = {:token => Base64.encode64(token), :payload => Base64.encode64(payload)}

f = File.open(ARGV[1],'w');
f.write(sealed.to_json);
f.close